### ISSUE TITLE
Bug: Clear causes select to lose focus which closes the dropdown

### ISSUE THREAD

**Issue by shyamal890** [ng-select/ng-select#1471]
Pressing `clear` causes ng-select to loose focus causing dropdown to close. 

**Reproducbile example**
https://stackblitz.com/edit/ng-select-clear-issue?file=main.ts

**To Reproduce** 
Steps to reproduce the behavior (if example is not provided):
1. Open the dropdown
2. Click on 'x'
3. It would close dropdown, when it should not.

**Expected behavior**
Clear should not close the dropdown

**Comment by chetanladdha** [comment]
Yes, with Angular 8 issue is reproducible from 3.0

**Comment by shyamal890** [comment]
Hi @varnastadeus can you please have a look at this. Thanks in advance.

**Comment by rp0m** [comment]
Having the same issue here in Angular 7

**Comment by svbackend** [comment]
In general select is being closed even when you're trying to scroll the screen to see it on mobile phone (android 11, chrome).

Reproduce:
1. Open select
2. It should show many options and some of them not visible because of keyboard at the bottom
3. Scroll down to see all the options, but not inside of the selectmenu, but a page scroll
4. Select is closed, even though you didn't click outside of it, just scrolled

Looks like it's closed on mouseDown or touchStart event, while I expect it to close on click

**Comment by jvegaseg** [comment]
Any advance on this issue?

**Comment by erkanpulat** [comment]
Is there a solution?

**Comment by erkanpulat** [comment]
Hello @varnastadeus, can you please have a look at this? Do you have any suggestions for a solution?

**Comment by ZinedineGithub** [comment]
Hello, any news for this issue? I'm having the same issue with my parent popover being closed

**Comment by kelvinsleonardo** [comment]
The same issue here. Any solution?

**Comment by rambalusu** [comment]
[closeOnSelect]="false"

**Comment by steimo** [comment]
I also faced a similar issue while using **ngbDropdown** and **ng-select**.

After some experimentation, I found a workaround that might help:

1. Make the **[autoClose]** attribute dynamic by introducing a **preventDropdownClose** variable:
```html
<div ngbDropdown container="body" [autoClose]="preventDropdownClose ? false : 'outside'">
...
</div>
```

2. Add a **(clear)** output event to your ng-select and create a new function:

```html
<ng-select
    bindLabel="name"
    bindValue="id"
    placeholder="All"
    (clear)="ngSelectOnClear()"
    [items]="items"
    [clearable]="true">
</ng-select>
```

```typescript
  ngSelectOnClear() {
    this.preventDropdownClose = true;

    setTimeout(() => this.preventDropdownClose = false, 10)
  }
```

Using a timeout prevents the dropdown from closing immediately after clearing the selection.

**Comment by pavankjadda** [comment]
Let me take a look at this

**Comment by pavankjadda** [comment]
After some digging I found out, `ng-select` clear works as expected. It's `ngbDropdown` that closing the dropdown by thinking option is selected. Since this is out of the scope for the project, I would recommend steimo [reply](https://github.com/ng-select/ng-select/issues/1471#issuecomment-2000206344)

**Comment by joostme** [comment]
We just came across the very same issue when using NgSelect in combination with NgBootstrap.
One solution I found is to add a `event.stopPropagation()` to the `handleMouseDown` method in [NgSelect](https://github.com/ng-select/ng-select/blob/9d04d1b1c8967b30cb62957c695f471307505e17/src/ng-select/lib/ng-select.component.ts#L396)

```ts
handleMousedown($event: MouseEvent) {
// ...
  if (target.classList.contains('ng-clear-wrapper')) {
          $event.stopPropagation(); // Like this
	  this.handleClearClick();
	  return;
  }
// ...
}
```

We did this by overriding the `handleMouseDown` method in our `ngAfterViewInit` method.

```ts
@ViewChild(NgSelectComponent, { static: true }) dropdown!: NgSelectComponent;
// ...
ngAfterViewInit() {
  const originalMethod = this.dropdown.handleMousedown.bind(this.dropdown);
  this.dropdown.handleMousedown = (event: MouseEvent) => {
    if ((event.target as HTMLElement).classList.contains('ng-clear-wrapper')) {
      event.stopPropagation();
    }
    return originalMethod(event);
  };
}

``` 
This prevents NgBootstrap from receiving this specific click event.

Would it be an option to add this to NgSelect? 
There is already a `$event.preventDefault()` in the `handleMouseDown` method ([see here](https://github.com/ng-select/ng-select/blob/9d04d1b1c8967b30cb62957c695f471307505e17/src/ng-select/lib/ng-select.component.ts#L396)), so this might also be a good place to add.

I am however aware, that this causes an unwanted breaking change for people, that rely on this event to be propagated (which I think is unlikely, but you never know).
